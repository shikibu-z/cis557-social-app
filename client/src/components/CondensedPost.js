/* eslint-disable no-unused-vars */
/* eslint-disable react/jsx-no-bind */
import { React, useEffect, useState } from 'react';
import { Card, Row, Col } from 'react-bootstrap';
import {
  Link,
} from 'react-router-dom';
import {
  Flag,
} from 'react-bootstrap-icons';
// import VoteBox from './VoteBox';
import FlagPost from './FlagPost';
import SpecificPost from '../pages/SpecificPost';

function CondensedPost(props) {
  const {
    group, title, info, isFlagged, postid,
  } = props;
  const userId = localStorage.getItem('currUser');
  const postId = postid.toString();
  const [flagged, setFlagged] = useState(isFlagged);
  const [modalShow, setModalShow] = useState(false);

  function handleClick() {
    if (!flagged) {
      setModalShow(true);
    }
  }

  useEffect(() => {
    if (flagged) {
      const element = document.getElementById(postId);
      element.style.color = '#F8B114';
    }
  }, [flagged]);

  return (
    <div>
      <Card style={{
        width: '55rem',
        height: '7rem',
      }}
      >
        <Row className="g-4">
          <Col xs={12} md={11}>
            <Card style={{ border: 'none' }}>
              <Card.Header style={{
                fontFamily: 'Roboto',
                fontSize: '1.5em',
                backgroundColor: 'white',
                border: 'none',
                fontWeight: 'bold',
              }}
              ><Link style={{ textDecoration: 'none', color: 'black' }} to={`/explore/${group}/${postId}`}>{title}</Link>

              </Card.Header>
            </Card>
            <Card style={{
              border: 'none',
              textAlign: 'right',
              position: 'absolute',
              bottom: '0',
              right: '0',
            }}
            >
              <Card.Text><Flag id={postId} className="flag" onClick={handleClick} /> {info}</Card.Text>
            </Card>
          </Col>
        </Row>
      </Card>
      <FlagPost
        show={modalShow}
        onHide={() => setModalShow(false)}
        handleFlag={() => { setFlagged(true); setModalShow(false); }}
        handleClose={() => setModalShow(false)}
        groupId={group}
        postId={postId}
        userId={userId}
      />
    </div>
  );
}

export default CondensedPost;
