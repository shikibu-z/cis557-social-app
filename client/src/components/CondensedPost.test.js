/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CondensedPost from './CondensedPost';

test('Condensed post matches snapshot', () => {
  const component = renderer.create(<BrowserRouter><CondensedPost group="test" title="testt" info="test" isFlagged={false} postid={1} /></BrowserRouter>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
