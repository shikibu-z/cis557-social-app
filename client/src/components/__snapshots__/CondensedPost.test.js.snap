// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Condensed post matches snapshot 1`] = `
<div>
  <div
    className="card"
    style={
      Object {
        "height": "7rem",
        "width": "55rem",
      }
    }
  >
    <div
      className="g-4 row"
    >
      <div
        className="col-md-11 col-12"
      >
        <div
          className="card"
          style={
            Object {
              "border": "none",
            }
          }
        >
          <div
            className="card-header"
            style={
              Object {
                "backgroundColor": "white",
                "border": "none",
                "fontFamily": "Roboto",
                "fontSize": "1.5em",
                "fontWeight": "bold",
              }
            }
          >
            <a
              href="/explore/test/1"
              onClick={[Function]}
              style={
                Object {
                  "color": "black",
                  "textDecoration": "none",
                }
              }
            >
              testt
            </a>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "border": "none",
              "bottom": "0",
              "position": "absolute",
              "right": "0",
              "textAlign": "right",
            }
          }
        >
          <p
            className="card-text"
          >
            <svg
              className="flag"
              fill="currentColor"
              height="1em"
              id="1"
              onClick={[Function]}
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.778.085A.5.5 0 0 1 15 .5V8a.5.5 0 0 1-.314.464L14.5 8l.186.464-.003.001-.006.003-.023.009a12.435 12.435 0 0 1-.397.15c-.264.095-.631.223-1.047.35-.816.252-1.879.523-2.71.523-.847 0-1.548-.28-2.158-.525l-.028-.01C7.68 8.71 7.14 8.5 6.5 8.5c-.7 0-1.638.23-2.437.477A19.626 19.626 0 0 0 3 9.342V15.5a.5.5 0 0 1-1 0V.5a.5.5 0 0 1 1 0v.282c.226-.079.496-.17.79-.26C4.606.272 5.67 0 6.5 0c.84 0 1.524.277 2.121.519l.043.018C9.286.788 9.828 1 10.5 1c.7 0 1.638-.23 2.437-.477a19.587 19.587 0 0 0 1.349-.476l.019-.007.004-.002h.001M14 1.221c-.22.078-.48.167-.766.255-.81.252-1.872.523-2.734.523-.886 0-1.592-.286-2.203-.534l-.008-.003C7.662 1.21 7.139 1 6.5 1c-.669 0-1.606.229-2.415.478A21.294 21.294 0 0 0 3 1.845v6.433c.22-.078.48-.167.766-.255C4.576 7.77 5.638 7.5 6.5 7.5c.847 0 1.548.28 2.158.525l.028.01C9.32 8.29 9.86 8.5 10.5 8.5c.668 0 1.606-.229 2.415-.478A21.317 21.317 0 0 0 14 7.655V1.222z"
              />
            </svg>
             
            test
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
